content/en/blog/_posts/2023-10-18-gateway-api-time-to-switch-i2gw/index.md

```diff
@@ -0,0 +1,203 @@
+Gateway API has a much broader base set of features and a standard pattern for extensions that can
+be more expressive than annotations were on Ingress.
+
+
+## Upgrading to Gateway
+
+Migrating from Ingress to the Gateway API may seem intimidating, but luckily Kubernetes just
+released a tool to simplify the process.
+[ingress2gateway](https://github.com/kubernetes-sigs/ingress2gateway) assists in the migration by
+converting your existing Ingress resources into Gateway API resources. Here is how you can get
+started with Gateway API and using ingress2gateway:
+
+1. [Install a Gateway
+   controller](https://gateway-api.sigs.k8s.io/guides/#installing-a-gateway-controller) OR [install
+   the Gateway API CRDs manually](https://gateway-api.sigs.k8s.io/guides/#installing-gateway-api) .
+
+2. Install [ingress2gateway](https://github.com/kubernetes-sigs/ingress2gateway).
+   
+   If you have a Go development environment locally, you can install `ingress2gateway` with:
+
+   ```
+   go install github.com/kubernetes-sigs/ingress2gateway@v0.1.0
+   ```
+
+   This installs `ingress2gateway` to `$(go env GOPATH)/bin/ingress2gateway`.
+
+   Alternatively, follow the installation guide
+   [here](https://github.com/kubernetes-sigs/ingress2gateway#installation).
+
+3. Once the tool is installed, you can use it to convert the ingress resources in your cluster to
+   Gateway API resources.
+
+   ```
+   ingress2gateway print
+   ```
+
+   This above command will:
+
+   1. Load your current Kubernetes client config including the active context, namespace and
+      authentication details.
+   2. Search for ingresses and provider-specific resources in that namespace.
+   3. Convert them to Gateway API resources (Currently only Gateways and HTTPRoutes). For other
+   options you can  run the tool with `-h`, or refer to
+   [https://github.com/kubernetes-sigs/ingress2gateway#options](https://github.com/kubernetes-sigs/ingress2gateway#options).
+
+4. Review the converted Gateway API resources, validate them, and then apply them to your cluster.
+
+5. Send test requests to your Gateway to check that it is working. You could get your gateway
+   address using `kubectl get gateway <gateway-name> -n <namespace> -o
+   jsonpath='{.status.addresses}{"\n"}'`.
+
+6. Update your DNS to point to the new Gateway. 
+
+7. Once you've confirmed that no more traffic is going through your Ingress configuration, you can
+   safely delete it.
+
+## Wrapping up
+
+Achieving reliable, scalable and extensible networking has always been a challenging objective. The
+Gateway API is designed to improve the current Kubernetes networking standards like ingress and
+reduce the need for implementation specific annotations and CRDs.
+
+It is a Kubernetes standard API, consistent across different platforms and implementations and most
+importantly it is future proof. The Gateway API is considered the successor to Ingress; Gateway API
```

Comment:
A couple nits here:

1. Always prefer "Gateway API" over "The Gateway API" (comment applies throughout, but will avoid adding suggestion everywhere)
2. We can't say "successor" here as that can be construed as "replacement", but we are not getting rid of the Ingress API.
```suggestion
importantly it is future proof. Gateway API is the next generation of the Ingress API, but has a larger scope than that, expanding to tackle mesh and L4 routing as well. Gateway API
```

Reply:
We don't have a good name for the family of API kinds related to Gateway, and this is a problem. How do docs writers separate Gateway the API kind from Gateway the family of APIs?

> Always prefer "Gateway API" over "The Gateway API" (comment applies throughout, but will avoid adding suggestion everywhere)

Can you PR that to https://kubernetes.io/docs/contribute/style/style-guide/ first? (I'm going to push back, though - using a definite article here feels more appropriate).

Reply:
Also, avoid “L4”; try “layer 4” instead.

Reply:
>How do docs writers separate Gateway the API kind from Gateway the family of APIs

Somehow I think we've generally managed to just refer to "Gateway" or "Gateway resource", for example https://gateway-api.sigs.k8s.io/api-types/gateway/. Naming is hard, but this project is first and foremost a set of APIs. For the record, the first name of the project was "Service APIs" and that was even worse.

>Can you PR that to https://kubernetes.io/docs/contribute/style/style-guide/ first? (I'm going to push back, though - using a definite article here feels more appropriate).

"Gateway API" is the name of the project. You wouldn't use "The Kubernetes". I could see using "The Gateway API" to refer to just the API for the Gateway resource, but it is decidedly not the name of the project. Unfortunately usage is currently mixed between "Gateway API" and "The Gateway API" (including in this blog post). If you feel strongly here, I can try to find some time to follow up with a PR to the style guide.

Reply:
Mmm, OK. We like to get docs more polished for beta (and for in-tree things we usually get those nudges to happen).

I recommend that the first use of _Gateway API_ should appear in italics. That then fits with the style guide's advice.

Reply:
Yeah, would be great to get Gateway API docs more polished. Definitely one of the many challenges of building an API out of tree. If there's anything sig-docs can help with on that front it would be very welcome.